### ISSUE TITLE
"syntax" block

### ISSUE THREAD

**Issue by WreckedAvent** [sweet-js/sweet.js#581]
Context:

I want to be able to write private helpers for my macros. Currently, the only way to do this is to have another module which you import. I suggest having a `syntax` block which puts all of the statements into the correct phase to be used in macros.

Example:

```javascript
syntax {
  let next = ctx => ctx.next().value
  syntax inc = ctx => #`${next(ctx)} + 1`
}

inc 10
```

would compile to

```javascript
10 + 1
```

Considerations:

* @jlongster mentioned a `forSyntax` accepting an IIFE to achieve this instead. This seems less intuitive to me.

* The `import ... for syntax` could be implied within a `syntax` block

* `syntax identifier` seems kind of awkward to me inside of a `syntax` block but not sure if there's any better ways to approach it

**Comment by disnet** [comment]
Yeah, racket calls this `begin-for-syntax` and we've discussed it before in the context of the initial [module implementation](https://github.com/sweet-js/sweet.js/pull/531#issuecomment-211281804).

I definitely want this but haven't found a syntax I'm happy with yet. Nesting like racket or your suggestion goes against normal binding forms (except for `var`s hoisting kinda).

Maybe a prefix on the binding?

```js
let x for syntax = // ...
for syntax let x = // ...
syntax let x = // ...
// maybe a special labeled statement?
syntax: let x = //..
```

**Comment by WreckedAvent** [comment]
Well, personally, if it isn't a block, it's going to be a lot of repeated stuff::

    import m1 from ... for syntax
    import m2 from ... for syntax
    for syntax let h1 ...
    for syntax let h2 ...

vs

    for syntax {
      import m1 from ...
      import m2 from ...
      let h1 ...
      let h2 ...
    }

**Comment by disnet** [comment]
Agreed. Seems like our options are either violate DRY or violate lexical
nesting. I don't like either.

**Comment by vendethiel** [comment]
I propose commas to separate the let / imports.

**Comment by WreckedAvent** [comment]
@vendethiel you mean like this?

```javascript
import m1 from ...,
import m2 from ... for syntax

for syntax let h1 ...,
           let h2 ...

**Comment by gabejohnson** [comment]
@disnet `var` violates lexical nesting so there's precedent. I personally like the block solution. It could be implemented later as a reader macro though.

**Comment by gabejohnson** [comment]
@WreckedAvent Now that I look at it again I like your solution if we could do:
```js
for syntax const foo ...;
for syntax const bar ...,
                 baz ...;
```
This is no different than how binding forms currently work. As far as the imports go though ¯\\\_(ツ)_/¯ In the common case you shouldn't have too many imports `for syntax`.
